sphinxcontrib/needs/directives/needtable.py

```diff
@@ -64,16 +64,20 @@ def run(self):
 
         sort = self.options.get("sort", "id_complete")
 
-        caption = None
         if self.arguments:
-            caption = self.arguments[0]
+            title_text = self.arguments[0]
+            text_nodes = self.state.inline_text(title_text, self.lineno)[0]
+            title = nodes.title(title_text, "", *text_nodes)
```

Comment:
I would try to keep the creating of nodes out of run() and only collect data here.

The creation then is done in process().
So need_all_needtables mostly contain only str and not more

Reply:
So please move the node creation to process()

Reply:
Done.